app/org/sagebionetworks/bridge/models/sharedmodules/SharedModuleMetadata.java

```diff
@@ -0,0 +1,106 @@
+package org.sagebionetworks.bridge.models.sharedmodules;
+
+import java.util.Set;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+import org.sagebionetworks.bridge.exceptions.InvalidEntityException;
+import org.sagebionetworks.bridge.hibernate.HibernateSharedModuleMetadata;
+import org.sagebionetworks.bridge.json.BridgeTypeName;
+import org.sagebionetworks.bridge.models.BridgeEntity;
+
+/** Metadata for shared modules. */
+@BridgeTypeName("SharedModuleMetadata")
+@JsonDeserialize(as = HibernateSharedModuleMetadata.class)
+public interface SharedModuleMetadata extends BridgeEntity {
+    /** Convenience method for creating a SharedModuleMetadata using a concrete implementation. */
+    static SharedModuleMetadata create() {
+        return new HibernateSharedModuleMetadata();
+    }
+
+    /** Module ID. */
+    String getId();
+
+    /** @see #getId */
+    void setId(String id);
+
+    /** True if usage of this module is restricted by license or copyright. */
+    boolean isLicenseRestricted();
+
+    /** @see #isLicenseRestricted */
+    void setLicenseRestricted(boolean licenseRestricted);
+
+    /** Whether this module is a schema or a survey. Throws if the module is neither. */
+    default SharedModuleType getModuleType() {
+        if (getSchemaId() != null) {
+            return SharedModuleType.SCHEMA;
+        } else if (getSurveyGuid() != null) {
+            return SharedModuleType.SURVEY;
+        } else {
+            throw new InvalidEntityException("Shared module has neither schema nor survey");
+        }
+    }
```

Comment:
I don’t think there’s a lot of value in adding tasks to module metadata. Tasks are just a pointed to one or more surveys and schemas, and they’re very easy to replicate, and adding the functionality to ensure referential integrity would add an immense amount of complexity to the system.

On the other hand, the biggest goal for shared modules is so researchers can set up, for example, Tapping Activity without having to figure out how to create the schemas themselves, or add UPDRS survey without having to create the survey themselves. In this sense, schemas and surveys are definitely the atomic unit of a module.